test/hotspot/jtreg/runtime/MemberName/ResolvedMethodTableHash.java

```diff
@@ -48,6 +51,13 @@ private MethodHandle generate(String className) throws ReflectiveOperationExcept
         return MethodHandles.publicLookup().findStatic(cls, "m", MethodType.methodType(void.class));
     }
 
+    private MethodHandle generateWithSameName() throws ReflectiveOperationException {
+        byte[] buf = new byte[100];
+        int size = writeClass(buf, "MH$$");
+        Class<?> cls = Unsafe.getUnsafe().defineAnonymousClass(ResolvedMethodTableHash.class, Arrays.copyOf(buf, size), null);
```

Comment:
Because here needs different class loader data to load the different classes with the same name, I think Unsafe API is a simple way to achieve this purpose.

Reply:
But as you note, the name mangling for hidden classes would mean that the names are not actually the same. And I don't see how this introduces a different classloader-data. ??

Reply:
Unsafe.getUnsafe().defineAnonymousClass will create an unsafe anonymous class rather than a hidden class, and for unsafe anonymous class, JVM will create a new CLD, see the comment in SystemDictionary::resolve_hidden_class_from_stream:
```
  // - for unsafe anonymous class: create a new CLD whith a class holder that uses
  //                               the same class loader as the unsafe_anonymous_host.
  // - for hidden classes that are not strong: create a new CLD that has a class holder and
  //                                           whose loader is the Lookup class's loader.
  // - for hidden class: add the class to the Lookup class's loader's CLD.

```

Reply:
> Okay but we should only be concerned about the behaviour of public 
> API's, so we should be testing and benchmarking those API's.

> defineAnonymousClass() maybe relevant for 11u, but not mainline.

I can use public API.
Another reason I used Unsafe API here is for saving metaspace.

Reply:
updated.